text/0052-resource-importing-support.md

```diff
@@ -0,0 +1,365 @@
+structure (as a JSON file) on command line using `--resource-mapping` argument.
+It maps the contruct tree path (in forward slash separated format) to the
+required resource identifier, for example:
+
+```json
+{
+  "MyApplicationStack/MyBucket/Resource": {
+    "BucketName": "foo-application-static"
+  },
+  "MyApplicationStack/Vpc": {
+    "VpcId": "vpc-123456"
+  }
+}
+```
+
+The resource mapping file can either be constructed manually or `cdk import`
+can be called with `--create-resource-mapping` to create the file using an
+interactive prompt. This allows two step workflow:
+
+1. User runs CDK CLI in interactive mode to create the mapping file to be
+   modified, peer-reviewed or checked in the version control
+2. CDK is run non-interactively to apply the changes (perform the import),
+   possibly by a CI/CD toolchain
+
+Resources that weren't identified automatically or by provided resource mapping
+will be omitted from import (skipped).
+
+CDK import logic considers a resource _identified_ as long as it has a non-empty
+value for all required identification properties. For CloudFormation import
+operation to succeed, the resource with such properties must both exist and must
+not belong to another CloudFormation stack. CDK will not perform any validation
+of the provided values - it will pass them to CloudFormation and in case of an
+error, propagates it back to the user.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @rix0rrr
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+New CDK CLI sub-command `cdk import` that works just like `cdk deploy` but for
+newly added constructs in the stack, instead of creating new AWS resources, it
+attempts to import corresponding existing resources.
+
+### Why should I use this feature?
+
+When an already existing AWS resource needs to be brought under CDK management
+the most convenient way of achieving this is using this feature (`cdk import`).
+
+There are serveral use cases for importing resources into CDK stacks:
+
+- **IaC adoption:** Resources were created manually using AWS Console / CLI
+  originally. As the project got more complex, the team opts for
+  infrastructure-as-code approach using CDK. To be able to organize the existing
+  resources in CDK stacks without recreation, the import feature is needed.
+- **CDK migration:** The team uses an infrastructure-as-code tool such as
+  Terraform, Pulumi or plain CloudFormation and chooses to migrate onto CDK. To
+  perform a smooth migration without resources recreation, it is important to:
+  1. make the original IaC orchestrator "abandon" the resource - remove it
+     from its orchestration without physically removing the resource
+  2. "import" the resource in a CDK stack
+- **App refactoring:** The team already uses CDK but a major refactor is needed
```

Comment:
This is good to identify as a use case. I like that you thought about it.

But in practice, we're going to have (in AWS) 20+ copies of every app deployed... so if you really want to refactor something across 20 copies, we're going to need more automation to take care of this.

In any case -- not for this RFC, we will deal with that when we get to it! Perhaps people will be clever and build something on top of the structured input file.

Reply:
added in the "possible future expansions" section